src/groups/mqb/mqbc/mqbc_storagemanager.cpp

```diff
@@ -349,15 +360,24 @@ void StorageManager::setPrimaryStatusForPartitionDispatched(
     }
     BSLS_ASSERT_SAFE(pinfo.primaryLeaseId() > 0);
 
+    const bmqp_ctrlmsg::PrimaryStatus::Value oldValue = pinfo.primaryStatus();
     BALL_LOG_INFO << d_clusterData_p->identity().description()
-                  << " Partition [" << partitionId << "]: "
-                  << "Setting the status of primary: "
+                  << " Partition [" << partitionId
+                  << "]: " << "Setting the status of primary: "
                   << pinfo.primary()->nodeDescription()
                   << ", primaryLeaseId: " << pinfo.primaryLeaseId()
-                  << ", from " << pinfo.primaryStatus() << " to " << value
-                  << ".";
+                  << ", from " << oldValue << " to " << value << ".";
 
     pinfo.setPrimaryStatus(value);
+    if (bmqp_ctrlmsg::PrimaryStatus::E_ACTIVE == value) {
+        d_fileStores[partitionId]->setPrimary(pinfo.primary(),
```

Comment:
why is this in `setPrimaryStatusForPartition` and not in `setPrimaryForPartition`?

Reply:
FileStore should only set primary after the primary becomes active. The FileStore is unable to work with a passive primary. I will rename the function as `FileStore::setActivePrimary` to make the point clear.